ext/phar/phar_internal.h

```diff
@@ -425,6 +425,17 @@ static inline enum phar_fp_type phar_get_fp_type(phar_entry_info *entry)
 	return PHAR_G(cached_fp)[entry->phar->phar_pos].manifest[entry->manifest_pos].fp_type;
 }
 
+static inline time_t source_date_epoch_time(time_t *tloc)
+{
+	const char *sde = getenv("SOURCE_DATE_EPOCH");
```

Comment:
This might not be safe, we probably need to expose a getenv API that acquires the environment lock.

Reply:
I've put up #6571 for that.

Reply:
Thanks, I finally seem to have worked out how to use this.